### ISSUE TITLE
Questions about distill_loss

### ISSUE THREAD

**Issue by haoren55555** [lucidrains/vit-pytorch#289]
sorry to bother, I see the distill_loss in distill.py as :
    distill_loss = F.kl_div(
                F.log_softmax(distill_logits / T, dim=-1),
                F.softmax(teacher_logits / T, dim=-1).detach(),
                reduction='batchmean')
I wonder why the teacher part uses the softmax function rather than log_softmax one, thanks.

**Comment by lucidrains** [comment]
@haoren55555 yeah, you could do log softmax for teacher too by setting `log_target = True` https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html . i'm just rolling with what pytorch offers

**Comment by vivekh2000** [comment]
That was insightful. However, in your implementation of class DistillMixin, you are passing the `cls` token from the MLP lead. 

The MLP head of the distillation token is implemented using LayerNorm, whereas the cls token MLP head is implemented in vit.py without LayerNorm. Why is it so? What am I missing? Please correct me if I am wrong.
![image](https://github.com/lucidrains/vit-pytorch/assets/90518409/ceb8628e-54d6-4686-83b7-2c039c07f772)

https://github.com/lucidrains/vit-pytorch/blob/90be7233a3f55c29692a72da6ee4dcb5aab267d4/vit_pytorch/vit.py#L111C14-L111C22

![image](https://github.com/lucidrains/vit-pytorch/assets/90518409/bac22254-3798-4a5d-9965-d7c75ec9d2d2)

**Comment by vivekh2000** [comment]
@lucidrains 
I have also noticed that in the paper, the authors mentioned that at test time, they fused the two heads, i.e., MLP heads for cls token and distilled token.
> for which we add the softmax output by the two classifiers to make the prediction. (p.8, paragraph 3)
> 

This fusion of two heads is not implemented in the forward method of the DistillMixin class. Please let me know if it is implemented elsewhere or if I am missing something. Thanks for the excellent codebase.
